controllers/MobileAppController.php

```diff
@@ -2,45 +2,23 @@
 
 namespace humhub\modules\fcmPush\controllers;
 
-use humhub\modules\admin\components\Controller;
-use humhub\modules\admin\notifications\NewVersionAvailable;
-use humhub\modules\fcmPush\models\FcmUser;
-use humhub\modules\user\models\User;
+use humhub\components\Controller;
+use humhub\libs\Html;
+use humhub\modules\fcmPush\helpers\MobileAppHelper;
 use Yii;
 
 class MobileAppController extends Controller
 {
-    /**
-     * Renders the index view for the module
-     *
-     * @return string
-     */
-    public function actionIndex()
+    public function actionInstanceOpener()
     {
-
-        if (Yii::$app->request->get('triggerNotification') == 1) {
-
-            /** @var User $user */
-            $user = Yii::$app->user->getIdentity();
-
-            $updateNotification = new NewVersionAvailable();
-            $updateNotification->sendBulk(User::find()->where(['user.id' => $user->id]));
-            $this->view->setStatusMessage('success', 'Notification queued!');
-            return $this->redirect('index');
-        }
-
-        if (Yii::$app->request->get('deleteToken') != "") {
-            $t = FcmUser::findOne(['id' => Yii::$app->request->get('deleteToken')]);
-            if ($t->delete() !== false) {
-                $this->view->setStatusMessage('success', 'Token deleted!');
-                return $this->redirect('index');
-            } else {
-                $this->view->setStatusMessage('warning', 'Token NOT deleted!');
-                return $this->redirect('index');
-            }
-        }
-
-        return $this->render('index');
+        // Send to the mobile app to display the instance opener
+        Yii::$app->session->set(MobileAppHelper::SESSION_VAR_SHOW_OPENER, 1);
+
+        // Stay on the same page, because when we come back from the mobile app to this instance
+        // Force full page refresh to trigger the onLayoutAddonInit event
+        return $this->renderContent(sprintf(
+            '<script ' . Html::nonce() . '>window.location.href = "%s";</script>',
```

Comment:
@luke- @yurabakhtin do you know a better solution to refresh the full page (avoiding pJax), and stay the current page (the one we were on just before clicking on "Switch Network"?
Or perhaps my code is fine?

Reply:
> @luke- @yurabakhtin do you know a better solution to refresh the full page (avoiding pJax), and stay the current page (the one we were on just before clicking on "Switch Network"? Or perhaps my code is fine?

I'm not sure exactly but something like the following should work?

```php
return Yii::$app->response->redirect(Yii::$app->request->referrer ?: Url::to(['some-view']));
```

Overall I'd probably do something like this;

```php
public function actionInstanceOpener()
{
    // Set session variable to indicate the instance opener should be displayed in the mobile app
    Yii::$app->session->set(MobileAppHelper::SESSION_VAR_SHOW_OPENER, 1);

    // Redirect back to the referring page or to a default route if no referrer exists
    $referrer = Yii::$app->request->referrer ?: Url::to(['/']);
    return Yii::$app->response->redirect($referrer);
}
```

Reply:
@ArchBlood thanks, but `Yii::$app->response->redirect()` doesn't refresh the hole page, it uses pJax.

Reply:
> @ArchBlood thanks, but `Yii::$app->response->redirect()` doesn't refresh the hole page, it uses pJax.

See https://github.com/humhub/fcm-push/pull/60/files#r1884053345

Reply:
@marc-farre Not sure, but maybe simply: 

```php
        return $this->htmlRedirect();
```

Reply:
@ArchBlood @luke- Thanks!
Yes, the combination of `'data-pjax' => '0',` and `return $this->htmlRedirect();` was a better option!
Done in commit https://github.com/humhub/fcm-push/pull/60/commits/ba00673d96591a7cad34817ffda24505664320be